BlockchainSdk/Blockchains/Sui/SuiTransactionBuilder.swift

```diff
@@ -0,0 +1,132 @@
+        
+        let signatureMock = Data(repeating: 0x01, count: 64)
+        
+        let compiled = try TransactionCompiler.compileWithSignatures(coinType: .sui,
+                                                                     txInputData: input.serializedData(),
+                                                                     signatures: signatureMock.asDataVector(),
+                                                                     publicKeys: publicKey.blockchainKey.asDataVector())
+        let output = try SuiSigningOutput(serializedData: compiled)
+        
+        return output.unsignedTx
+    }
+    
+    func buildForSign(transaction: Transaction) throws -> Data {
+        guard let suiFeeParameters = transaction.fee.parameters as? SuiFeeParameters else {
+            throw WalletError.failedToBuildTx
+        }
+        
+        let input = try input(amount: transaction.amount.value, destination: transaction.destinationAddress, fee: suiFeeParameters)
+        
+        let preImageHashes = try TransactionCompiler.preImageHashes(coinType: .sui, txInputData: input.serializedData())
+        let preSigningOutput = try TxCompilerPreSigningOutput(serializedData: preImageHashes)
+        
+        if preSigningOutput.error != .ok {
+            Log.debug("SuiPreSigningOutput has a error: \(preSigningOutput.errorMessage)")
+            throw WalletError.failedToBuildTx
+        }
+        
+        return preSigningOutput.dataHash
+        
+    }
+    
+    func buildForSend(transaction: Transaction, signature: Data) throws -> (txBytes: String, signature: String) {
+        guard let suiFeeParameters = transaction.fee.parameters as? SuiFeeParameters else {
+            throw WalletError.failedToBuildTx
+        }
+        
+        let input = try input(amount: transaction.amount.value, destination: transaction.destinationAddress, fee: suiFeeParameters)
+        
+        let compiled = try TransactionCompiler.compileWithSignaturesAndPubKeyType(coinType: .sui,
+                                                                                  txInputData: input.serializedData(),
+                                                                                  signatures: signature.asDataVector(),
+                                                                                  publicKeys: publicKey.blockchainKey.asDataVector(),
+                                                                                  pubKeyType: .ed25519)
+        
+        let output = try SuiSigningOutput(serializedData: compiled)
+        return (output.unsignedTx, output.signature)
+    }
+    
+    private func input(amount: Decimal, destination: String, fee: SuiFeeParameters) throws -> WalletCore.SuiSigningInput {
+        let decimalAmount = amount * decimalValue
+        let useCoins = getCoins(for: decimalAmount + fee.gasBudget)
```

Comment:
nit:  кажется такое название понятнее будет, не настаиваю.
```suggestion
        let сoinsToUse = getCoins(for: decimalAmount + fee.gasBudget)
```